raft/raft.go

```diff
@@ -434,6 +448,7 @@ func (r *raft) reset(term uint64) {
 		}
 	}
 	r.pendingConf = false
+	r.readOnly = newReadOnly(r.readOnly.option)
```

Comment:
only leader can do efficient l-read. if we want to retry, we have to rely on the previous leader to send the previous read info to the new leader. i feel this is worse than a simple application layer retry.